python-checks/src/main/java/org/sonar/python/checks/CheckUtils.java

```diff
@@ -163,4 +182,30 @@ public static boolean isAbstract(FunctionDef funDef) {
       .map(decorator -> TreeUtils.decoratorNameFromExpression(decorator.expression()))
       .anyMatch(foundDeco -> ABC_ABSTRACTMETHOD_DECORATORS.stream().anyMatch(abcDeco -> abcDeco.equals(foundDeco)));
   }
+
+  /**
+   * Simple check whether the given expression is the "self" name expression.
+   * It does not check whether the name is actually referring to a method parameter!
+   */
```

Comment:
You are absolutely right, doing a manual string comparison here is not the best approach when we can check for the symbol.

`isSelf` and `findSelfParameterSymbol` are actually methods from existing checks that I factored out, added tests for, and reused. I didn't want to change them too much so that I do not affect the behavior of the other checks.

I will try to add the TODO note, however, I believe the SonarQube code analysis check on GitHub will complain about new TODO comments in Java code. I'll see what happens.

Reply:
I think we can accept the code smell related to the TODO comment, as this is intentional :)